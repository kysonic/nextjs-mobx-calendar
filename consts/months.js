export const DAYS = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
export const MONTHS = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
export const DAYS_IN_MONTH = 31;
export const DAYS_IN_WEEK = 7;
