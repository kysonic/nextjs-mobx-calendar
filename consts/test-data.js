export const APP_INITIAL_DATA = {
    navigationStore: {
        items: [
            {
                "id": "home",
                "title": "Home",
                "href": "/"
            }
        ]
    },
    dateStore: {
        ISODate: '2019-01-01T00:00:00.000Z'
    }
};

export const CALENDAR_INITIAL_DATA = {
    "2019-04-10T00:00:00.000Z": [
        {
            "id": "abcde1",
            "title": "Action 1",
            "description": "Description of action 1"
        },
        {
            "id": "abcde2",
            "title": "Action 2",
            "description": "Description of action 2"
        }
    ],
    "2019-03-11T00:00:00.000Z": [
        {
            "id": "abcde2",
            "title": "Action 2",
            "description": "Description of action 2"
        },
        {
            "id": "abcde3",
            "title": "Action 4",
            "description": "Description of action 4"
        }
    ]
}
