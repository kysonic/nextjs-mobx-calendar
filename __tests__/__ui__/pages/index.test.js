import IndexPage from '../../../pages';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react/index';
import AppStore from '../../../stores/app';
import {APP_INITIAL_DATA, TEXT_DATA} from '../../../consts/test-data';

import axios from 'axios/index';

jest.mock('axios');
axios.get.mockResolvedValue({data: TEXT_DATA});

describe('Index page tests', () => {
    test('Renders without crashes', () => {
        const appStore = new AppStore(APP_INITIAL_DATA);
        const tree = renderer.create(
            <Provider appStore={appStore}>
                <IndexPage initialState={ [{text: TEXT_DATA.text}] } />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

