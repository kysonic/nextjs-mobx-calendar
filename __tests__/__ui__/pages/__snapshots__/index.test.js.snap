// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page tests Renders without crashes 1`] = `
<div
  className="jsx-3569970921"
>
  <nav
    className="jsx-3989381672"
  >
    <ul
      className="jsx-3989381672"
    >
      <li
        className="jsx-3989381672"
      >
        <span
          onClick={[Function]}
        >
          Home
        </span>
      </li>
    </ul>
    <style>
      
                     ul li:hover .sub-items {
                        display: block;
                     }
                     ul li a, ul li a:visited, ul li a:active {
                        color: #666;
                     }
                     ul li:hover a {
                        color: #444;
                     }
                
    </style>
  </nav>
  <div
    className="jsx-3569970921 content"
  >
    <div
      className="jsx-2395214420 text"
      data-testid="text"
    >
      Welcome to the Mobx Calendar
    </div>
  </div>
</div>
`;
